jax/interpreters/parallel.py

```diff
@@ -169,30 +169,45 @@ def pack(self, tracers):
 
 newvar = pe.gensym('_axis')
 
-def papply(fun, name, in_vals, in_axes):
-  return papply_transform(fun).call_wrapped(name, in_vals, in_axes)
+def papply(fun, name, in_vals, axis_size, in_axes, out_axis):
+  out_val = papply_transform(fun).call_wrapped(
+      name, in_vals, axis_size, in_axes, out_axis)
+  return out_val
+
+def _moveaxis(x, src, dst):
```

Comment:
I think this doesn't handle tuples. Can you just call `batching.moveaxis`?

Also, in the `src is None` case, if `dst` is not also None should that be an error or something? (That's a case where `pmap`/`vmap` would broadcast, hence why `batching.moveaxis` takes a `sz` argument.)